cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -217,6 +218,29 @@ public void getArtifactInfo(HttpRequest request, HttpResponder responder,
   }
 
   @GET
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/parents")
```

Comment:
I don't think we need a new endpoint, but can simply extend the getArtifactInfo one to include the parents in the response. In other words, add a `Set<ArtifactRange> parents` field to ArtifactInto.